robot/impl/local_robot.go

```diff
@@ -684,27 +689,44 @@ func (r *localRobot) updateWeakDependents(ctx context.Context) {
 	// service depends on all resources.
 	// For now, we pass all resources and empty configs.
 	for resName, res := range internalResources {
-		switch resName {
-		case web.InternalServiceName:
-			if err := res.Reconfigure(ctx, allResources, resource.Config{}); err != nil {
-				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
-			}
-		case framesystem.InternalServiceName:
-			fsCfg, err := r.FrameSystemConfig(ctx)
-			if err != nil {
-				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
-				continue
-			}
-			if err := res.Reconfigure(ctx, components, resource.Config{ConvertedAttributes: fsCfg}); err != nil {
-				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
+		resChan := make(chan struct{}, 1)
+		resName := resName
```

Comment:
@edaniels more reason to enable the [loopvar](https://github.com/golang/go/wiki/LoopvarExperiment) experiment.

Reply:
hah yes